### PR TITLE
chore: add CodeQL code scanning

### PR DESCRIPTION
**Pull request by stijnmoreels**
Add CodeQL code scanning.

Relates to https://github.com/arcus-azure/arcus/discussions/195

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
+    - cron: '29 5 * * 4'
```
Comment (fgheysels) [inline .github/workflows/codeql-analysis.yml:21]: Just out of curiousity; why do we execute this code scanning on a schedule basis as well ?  If you do it on every completed PR on master, I don't see any necessity do to this on a schedule basis as well ?

**Review by fgheysels — APPROVED** [review]
I like it that even in test-scenario's we have no hardcoded username/passwords in our code. :)

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (stijnmoreels) [reply .github/workflows/codeql-analysis.yml:21]: Hmm, yeah, that is correct. Hmm, maybe this is bc the CodeQL scanning database is getting updated outside PR's. So, if a new security vulnerability surfaces, we don't have to wait for someone to make a PR to receive the notification.

Comment (fgheysels) [reply .github/workflows/codeql-analysis.yml:21]: Good point.

**Review by fgheysels — APPROVED** [review]

**Review by fgheysels — APPROVED** [review]

**Pull request merged**